enos/modules/upgrade_servers/main.tf

```
@@ -42,24 +42,26 @@ resource "enos_local_exec" "wait_for_leader" {
   scripts = [abspath("${path.module}/scripts/wait_for_stable_cluster.sh")]
 }
 
-////////////////////////////////////////////////////////////////////////////////
-//                    Upgrading the first server
-////////////////////////////////////////////////////////////////////////////////
-// Taking a snapshot forces the cluster to store a new snapshot that will be 
-// used to restore the cluster after the restart, because it will be the most 
-// recent available, the resulting file wont be used..
-resource "enos_local_exec" "take_first_cluster_snapshot" {
+// Forcing a snapshot from the leader drives the cluster to store the most recent
+// state and exercise the snapshot restore at least once when upgrading.
+// The resulting file wont be used. 
+// The stale flag defaults to "false" but it is included to reinforce the fact
+// that it has to be taken from the leader for future readers.
+resource "enos_local_exec" "take_cluster_snapshot" {
   depends_on = [enos_local_exec.wait_for_leader]
 
   environment = local.nomad_env
 
   inline = [
-    "nomad operator snapshot save -stale -address https://${var.servers[0]}:4646 ${random_pet.upgrade.id}-0.snap",
+    "nomad operator snapshot save --stale false ${random_pet.upgrade.id}-0.snap",
```

Our command line parser is really weird for stuff like this:

```suggestion
    "nomad operator snapshot save -stale=false ${random_pet.upgrade.id}-0.snap",
```

```
$ nomad operator snapshot save --stale false /tmp/snapshot1.snap
This command takes either no arguments or one: <filename>
For additional help try 'nomad operator snapshot save -help'

$ nomad operator snapshot save --stale=false /tmp/snapshot1.snap
State file written to /home/tim/tmp/snapshot1.tar.gz
```